rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -1035,6 +1039,9 @@ def _get_logs(
                             'Couldnt decode an etherscan event due to {str(e)}}',
                         ) from e
 
+                if callback is not None:
```

Comment:
1. This is not enough. You are only doing this for the etherscan case. All other nodes go through `_query_web3_get_logs` which you can see in the very start.
2. This function is huge and hard to read. Probably why you missed it. Perhaps break it down in another commit?

Reply:
I didn't abstract this logic now. I prefer to do it in another moment